x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java

```diff
@@ -298,8 +300,31 @@ private IO(final int sharedBytesPos, MappedByteBuffer mappedByteBuffer) {
             this.mappedByteBuffer = mappedByteBuffer;
         }
 
+        @Override
+        public boolean tryIncRef() {
+            return SharedBytes.this.tryIncRef();
```

Comment:
I think this means all reads and writes will now have one ref-count that all threads contend on. This seems sub optimal relative to having something more optimistic?

Reply:
Agree it is suboptimal but I haven't found a better way to protect read/write access.

Reply:
I reduced a bit the contention by having each IO using its own refcount in ab1351f215341254f091b26376aa4fd6288841c8